lib/sles4sap/sap_deployment_automation_framework/inventory_tools.pm

```diff
@@ -0,0 +1,220 @@
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::inventory_tools;
+
+use warnings;
+use strict;
+use YAML::PP;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use sles4sap::sap_deployment_automation_framework::naming_conventions qw(deployer_private_key_path sut_private_key_path);
+use sles4sap::console_redirection;
+
+=head1 SYNOPSIS
+
+Library contains functions that handle SDAF inventory file.
+
+SDAF inventory yaml file example:
+QES_DB:
+  hosts:
+    dbhost01:
+      ansible_host        : 192.168.1.2
+      ansible_user        : someuser
+      ansible_connection  : ssh
+      connection_type     : key
+      virtual_host        : virtualhostname01
+      become_user         : root
+      os_type             : linux
+      vm_name             : somevmname01
+    dbhost02:
+      ansible_host        : 192.168.1.3
+      ansible_user        : someuser
+      ansible_connection  : ssh
+      connection_type     : key
+      virtual_host        : virtualhostname02
+      become_user         : root
+      os_type             : linux
+      vm_name             : somevmname02
+  vars:
+    node_tier             : hana
+    supported_tiers       : [hana, scs, pas]
+QES_SCS:
+  hosts:
+  vars:
+    node_tier             : scs
+    supported_tiers       : [scs, pas]
+QES_ERS:
+  hosts:
+  vars:
+    node_tier             : ers
+    supported_tiers       : [ers]
+
+=cut
+
+our @EXPORT = qw(
+  read_inventory_file
+  prepare_ssh_config
+  verify_ssh_proxy_connection
+);
+
+=head2 read_inventory_file
+
+    read_inventory_file($sap_inventory_file_path);
+
+Returns full path to an existing ansible inventory file
+
+=over
+
+=item * B<inventory_file_path> Full file path pointing to SDAF inventory file
+
+=back
+=cut
+
+sub read_inventory_file {
+    my ($inventory_file_path) = @_;
+    my $ypp = YAML::PP->new;
+    my $raw_file = script_output("cat $inventory_file_path");
+    my $yaml_data = $ypp->load_string($raw_file);
+    return $yaml_data;
+}
+
+=head2 ssh_config_entry_add
+
+    ssh_config_entry_add(entry_name=$entry_name, hostname=>$hostname
+        [, identity_file=>$identity_file, identities_only=><bool>, proxy_jump=$proxy_jump, batch_mode=>bool]);
+
+Add host entry into ~/.ssh/config
+
+=over
+
+=item * B<entry_name> Config entry name. This name can be used instead of host/IP in ssh command. Example: ssh root@<entry_name>
+
+=item * B<user> Define ssh username
+
+=item * B<hostname> Target hostname or IP addr
+
+=item * B<identity_file> Full path to SSH private key
+
+=item * B<identities_only> If true, SSH will only attempt passwordless login
+
+=item * B<batch_mode> If true, all SSH interactive features will be disabled. Test won't have to wait for timeouts.
+
+=item * B<proxy_jump> Jump host hostname, IP addr or point to another entry in config file
+
+=item * B<strict_host_key_checking> Turn off host key check
+
+=back
+=cut
+
+sub ssh_config_entry_add {
```

Comment:
This function is so beautifully generic that should live in a more generic place. https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/utils.pm or something https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/lib/sles4sap

Reply:
yes, was just an idea